pageserver/src/tenant/tasks.rs

```diff
@@ -102,6 +102,7 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
             let started_at = Instant::now();
 
             let sleep_duration = if period == Duration::ZERO {
+                #[cfg(feature = "testing")]
```

Comment:
Actually I think we need

```suggestion
                #[cfg(not(feature = "testing"))]
```

To have these nagging us in prod, but not while doing (manual) testing.

Reply:
These could be even refactored to a method similar to `warn_when_period_overrun` with similar or equal to "task" argument, then we'd have a single place to log this "we don't want the repeated log line in tests, but we want it in prod".

Reply:
Thanks for the tip! @koivunej 
I think if there are more cases, we can add helper method similar to `warn_when_period_overrun`.